library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java

```diff
@@ -110,15 +108,14 @@ public void run() {
 
     private void handleNotificationBroadcast(Context context, Intent intent) {
         String action = intent.getAction();
+        long[] ids = intent.getLongArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS);
+        int[] statuses = intent.getIntArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES);
         switch (action) {
             case ACTION_LIST:
-                long[] ids = intent.getLongArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS);
-                int[] statuses = intent.getIntArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES);
                 sendNotificationClickedIntent(context, ids, statuses);
                 break;
             case ACTION_OPEN: {
```

Comment:
yeah I see your point and makes sense to me as well, but I don't want to change everything in this PR and I'm not 100% sure of what the new names would be (since there are many other actions).
I'll create an issue to track this if you're ok with it?